packages/grid/x-data-grid/src/components/GridRow.tsx

```diff
@@ -256,141 +255,49 @@ const GridRow = React.forwardRef<HTMLDivElement, GridRowProps>(function GridRow(
-
-        const params: GridRenderEditCellParams = {
-          ...cellParams,
-          row: updatedRow,
-          ...editCellStateRest,
-          api: apiRef.current,
-        };
-
-        content = column.renderEditCell(params);
-        // TODO move to GridCell
-        classNames.push(clsx(gridClasses['cell--editing'], rootClasses?.['cell--editing']));
-      }
-
-      if (getCellClassName) {
-        // TODO move to GridCell
-        classNames.push(getCellClassName(cellParams));
-      }
+  const getCell = (
+    column: GridStateColDef,
+    cellProps: Pick<
+      GridCellProps,
+      'width' | 'colSpan' | 'showRightBorder' | 'indexRelativeToAllColumns'
+    >,
+  ) => {
+    const disableDragEvents =
+      (disableColumnReorder && column.disableReorder) ||
+      (!rowReordering &&
+        !!sortModel.length &&
+        treeDepth > 1 &&
+        Object.keys(editRowsState).length > 0);
+
+    return (
+      <CellComponent
+        key={column.field}
+        column={column}
+        width={cellProps.width}
+        rowId={rowId}
+        height={rowHeight}
+        showRightBorder={cellProps.showRightBorder}
+        align={column.align || 'left'}
+        colIndex={cellProps.indexRelativeToAllColumns}
+        colSpan={cellProps.colSpan}
+        disableDragEvents={disableDragEvents}
+        {...slotProps?.cell}
+      />
+    );
+  };
 
-      const hasFocus = focusedCell === column.field;
-      const tabIndex = tabbableCell === column.field ? 0 : -1;
-
-      const isSelected = apiRef.current.unstable_applyPipeProcessors('isCellSelected', false, {
-        id: rowId,
-        field: column.field,
-      });
-
-      return (
-        <CellComponent
-          key={column.field}
-          value={cellParams.value}
-          field={column.field}
-          width={cellProps.width}
-          rowId={rowId}
-          height={rowHeight}
-          showRightBorder={cellProps.showRightBorder}
-          formattedValue={cellParams.formattedValue}
-          align={column.align || 'left'}
-          cellMode={cellParams.cellMode}
-          colIndex={cellProps.indexRelativeToAllColumns}
-          isEditable={cellParams.isEditable}
-          isSelected={isSelected}
-          hasFocus={hasFocus}
-          tabIndex={tabIndex}
-          className={clsx(classNames)}
-          colSpan={cellProps.colSpan}
-          disableDragEvents={disableDragEvents}
-          {...slotProps?.cell}
-        >
-          {content}
-        </CellComponent>
-      );
-    },
-    [
-      apiRef,
-      rowId,
-      disableColumnReorder,
-      rowReordering,
-      sortModel.length,
-      treeDepth,
-      editRowsState,
-      getCellClassName,
-      focusedCell,
-      tabbableCell,
-      CellComponent,
-      rowHeight,
-      slotProps?.cell,
-      rootClasses,
-    ],
+  const sizes = useGridSelector(
+    apiRef,
+    () => ({ ...apiRef.current.unstable_getRowInternalSizes(rowId) }),
```

Comment:
I can confirm that this is required, some tests fail if we don't unwrap them. Because they are mutated directly, `fastShallowCompare`'s `a === b` check succeeds.